src/components/dialogs/StaffPaymentDialog.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <v-dialog ref="dialog" v-model="dialog" width="45rem" persistent :attach="attach"
+    content-class="staff-payment-dialog"
+  >
+    <v-card>
+      <v-card-title class="d-flex justify-space-between">
+        <div>Staff Payment</div>
+        <v-btn icon large class="dialog-close float-right" @click="exit()">
+          <v-icon>mdi-close</v-icon>
+        </v-btn>
+      </v-card-title>
+
+      <v-card-text>
+        <StaffPayment
+          :staffPaymentData="staffPaymentData"
+          :validate="validate"
+          :displaySideLabel="false"
+          :displayPriorityCheckbox="true"
+          @update:staffPaymentData="emitStaffPaymentData($event)"
+          @valid="staffPaymentFormValid=$event"
+        />
+      </v-card-text>
+
+      <v-card-actions class="justify-center">
+        <v-btn
+          id="dialog-close-button"
+          class="button-blue py-5 px-5"
+          :loading="loading"
+          :disabled="loading"
+          @click="exit()"
+        >
+          Exit Payment
+        </v-btn>
+        <v-btn
+          id="dialog-submit-button"
+          class="primary py-5 px-10"
+          :loading="loading"
+          :disabled="loading"
+          @click="submit()"
+        >
+          Submit
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Vue } from 'vue-property-decorator'
+import { StaffPayment } from '@/components/common'
+import { StaffPaymentIF } from '@/interfaces'
+import { StaffPaymentOptions } from '@/enums'
+
+@Component({
+  components: { StaffPayment }
+})
+export default class StaffPaymentDialog extends Vue {
+  // Refs
+  $refs!: {
+    dialog: any
+  }
+
+  // Prop to provide attachment selector.
+  @Prop() readonly attach: string
+
+  // Prop to display the dialog.
+  @Prop() readonly dialog: boolean
+
+  // Prop to provide staff payment data.
+  @Prop({ required: true }) readonly staffPaymentData: StaffPaymentIF
+
+  // Prop to display loading indicators and to disable buttons.
+  @Prop({ default: false }) readonly loading: boolean
+
+  private validate = false
+  private staffPaymentFormValid = false
+
+  /** Called when user clicks Submit button. */
+  private async submit (): Promise<void> {
+    // enable validation
+    this.validate = true
+
+    // wait for validation
+    await Vue.nextTick()
+
+    // emit submit event only if form is valid
+    // otherwise user stays on this dialog
+    if (this.staffPaymentFormValid) {
+      this.$emit('submit')
+    } else if (this.staffPaymentData.option === StaffPaymentOptions.NONE) {
+      // "bounce" the dialog to indicate error: user needs to select
+      // an option (when an option is selected, other errors display)
+      this.$refs.dialog.animateClick()
```

Comment:
fancy :)

Reply:
Hehe. This is the same animation used when the user clicks outside a modal. I felt something was needed here and this seems a good fit.